pontoon/pretranslation/utils.py

```diff
@@ -0,0 +1,70 @@
+from pontoon.machinery.utils import google_translate_util, translation_memory_util
+# from pontoon.base.models import Entity, Locale, Translation, TranslatedResource, User
+from django.utils import timezone
+import pontoon.base as base
+
+
+def pre_translate(project, locales=None, entities=None):
+    """
+    Identifies strings without any translations and any suggestions.
+    Engages TheAlgorithm (bug 1552796) to gather pre-translations.
+    Stores pre-translations as suggestions (approved=False) to DB.
+    Author should be set to a name like "Pontoon Translator <pontoon@mozilla.com>".
+    Stats and Latest activity should be updated
+
+    """
+
+    if not locales:
+        locales = project.locales.filter(project_locale__readonly=False)
+
+    if not entities:
+        entities = base.models.Entity.objects.filter(
+            resource__project=project,
+            obsolete=False,
+            translation__isnull=True,
+        )
+
+    now = timezone.now()
+
+    resources = project.resources.filter(obsolete=False)
+
+    user = base.models.User.objects.get(pk=2)
```

Comment:
That's too hardcoded. :) According to the spec, we should create a "Pretranslation" User.